grails-app/controllers/mama/ng/scheduler/ScheduleController.groovy

```diff
@@ -0,0 +1,117 @@
+package mama.ng.scheduler
+
+import grails.converters.JSON
+
+class ScheduleController {
+
+    static allowedMethods = [save: "POST", update: "PUT", delete: "DELETE"]
+
+    def cronParserService
+
+    def list() {
+        params.max = Math.min(params.max ? params.int('max') : 50, 200)
+        def list = Schedule.list(params)
+        withFormat {
+            json {
+                response.status = 200
+                render list as JSON
+            }
+        }
+    }
+
+    def show(Schedule instance) {
+        withFormat {
+            json {
+                response.status = 200
+                render instance as JSON
+            }
+        }
+    }
+
+    def messages(Schedule instance) {
+        def list = Message.findAllBySchedule(instance)
+        withFormat {
+            json {
+                response.status = 200
+                render list as JSON
+            }
+        }
+    }
+
+    def create(Schedule instance) {
+        instance.nextSend = cronParserService.determineNextDate(instance.cronDefinition)
```

Comment:
When creating a schedule I think it makes sense to be able to write things like ``create a schedule that fires every Monday after the expected date of birth`` rather than just ``create a schedule that fires every Monday after today``. I.e. I think it would be useful to be able to pass a custom date for ``determineNextDate`` to determine the next date relative to.